tuf/tuf.go

```diff
@@ -173,23 +162,133 @@ func (tr *Repo) RemoveBaseKeys(role string, keyIDs ...string) error {
 	return nil
 }
 
+// GetBaseRole gets a base role from this repo's metadata
+func (tr *Repo) GetBaseRole(name string) (data.BaseRole, error) {
+	if !data.ValidRole(name) {
+		return data.BaseRole{}, data.ErrInvalidRole{Role: name, Reason: "invalid base role name"}
+	}
+	if tr.Root == nil {
+		return data.BaseRole{}, ErrNotLoaded{data.CanonicalRootRole}
+	}
+	roleData, ok := tr.Root.Signed.Roles[name]
+	if !ok {
+		return data.BaseRole{}, data.ErrInvalidRole{Role: name, Reason: "role not found in root file"}
+	}
+	// Get all public keys for the base role from TUF metadata
+	keyIDs := roleData.KeyIDs
```

Comment:
@diogomonica previously, KeyDB refused to add a role if the keys IDs were not of keys already in the KeyDB.  Calling `SetRoot` or `SetTargets` before added the keys, then added the roles they defined, so that's where the check was before.